### PR TITLE
examples: Add locality_weighted_lb_config to locality-load-balancing example

### PR DESCRIPTION
**Pull request by liamhazell**
Commit Message: Add locality_weighted_lb_config to locality-load-balancing example
Additional Description: For the load_balancing_weight option to be used, this option must be set. Discussed in https://github.com/envoyproxy/envoy/issues/25692
Risk Level: None
Testing:
Docs Changes: 
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

Wasn't sure if there was a good spot to update the docs. The page on the locality weighted load balancer does describe that you need to set this [field](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/locality_weight#arch-overview-load-balancing-locality-weighted-lb:~:text=Locality%20weighted%20load%20balancing%20is%20configured%20by%20setting%20locality_weighted_lb_config%20in%20the%20cluster), it was just user error in not knowing that you need to have the `{}` present. :)

### CODE REVIEW

File: examples/locality-load-balancing/envoy-proxy.yaml [hunk 1]
```diff
@@ -39,6 +39,8 @@ static_resources:
   - name: backend
     type: STRICT_DNS
     lb_policy: ROUND_ROBIN
+    common_lb_config:
+      locality_weighted_lb_config: {} # Must be set for load_balancing_weight to be used
```
Comment (phlax) [inline examples/locality-load-balancing/envoy-proxy.yaml:43]: i would avoid adding this here - can we add something in the actual example docs here - https://github.com/envoyproxy/envoy/blob/main/docs/root/start/sandboxes/locality_load_balancing.rst (arggh the github ui is useless here)

we probably want to make this a `.. note::` somewhere to draw attention to it

ie

```rst

... note::
    The ``locality_weighted_lb_config`` must be set in ``common_lb_config``
    for the ``load_balancing_weight`` to be used.

```

or similar

**Review by phlax — COMMENTED** [review]
brilliant, thanks for following up on this @liamhazell

File: examples/locality-load-balancing/envoy-proxy.yaml (hunk 1 above)
Comment (phlax) [reply examples/locality-load-balancing/envoy-proxy.yaml:43]: currently rendered page https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/locality_load_balancing

probably we could put it at the end of the first section, just after where it mentions the config

Comment (liamhazell) [reply examples/locality-load-balancing/envoy-proxy.yaml]: Fixed, but I messed up the DCO trying to use my full name let me fix that

**Comment by liamhazell** [comment]
Also @phlax, would you be open to having an example contributed for the built in zone-aware routing? I see there currently isn't one, however I was messing around with that too and think it might be useful to have. Happy to try and contribute that when I have time

**Comment by phlax** [comment]
>  would you be open to having an example contributed for the built in zone-aware routing? 

very much so - scanning the docs and it looks like the kinda complexity that is best described by example

**Review by phlax — APPROVED** [review]
lgtm, thanks @liamhazell 

would be good to demonstrate the weights in this sandbox, altho its already quite complicated

**Comment by liamhazell** [comment]
> would be good to demonstrate the weights in this sandbox, altho its already quite complicated

Yeah thought this would be best left as an exercise for the reader to mess around with the settings


> very much so - scanning the docs and it looks like the kinda complexity that is best described by example

cool let me see what I can do!

**Pull request merged**